### PR TITLE
patch vite internally to get rid of the FSWatcher type error

### PR DESCRIPTION
**Pull request by freben**
Before:

https://github.com/backstage/backstage/actions/runs/8306808033/job/22735334479?pr=23239#step:12:10

Screenshot 2024-03-16 at 16 35 03

See https://github.com/vitejs/vite/issues/15714

### CODE REVIEW

**Comment by benjdlambert** [comment]
Can we just bump to 5 instead of the patches? It's fixed in v5 right?

**Comment by freben** [comment]
If you can. It's nontrivial I thought.

**Review by Rugvip — APPROVED** [review]

File: .changeset/renovate-376f928.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend-module-yeoman': patch
+---
+
+Updated dependency `yeoman-environment` to `^4.0.0`.
```
[the remark this answers is not in the document]
Comment (freben) [reply .changeset/renovate-376f928.md:5]: Myeah sorry I should have called it out in the PR description. Talked about it in person earlier with @benjdlambert - what the PR does unblocks that yeoman upgrade that dependabot has been failing to get through so it's sort of there as "proof" / a test.

**Review by vinzscam — APPROVED** [review]

**Comment by freben** [comment]
not sure this is even needed anymore

**Pull request closed without merging**